pkg/splunk/enterprise/afwscheduler.go

```diff
@@ -1052,27 +1120,40 @@ func (shcPlayBookContext *SHCPlayBookContext) runPlayBook() error {
 	appDeployContext := shcPlayBookContext.afwPipeline.appDeployContext
 
 	switch appDeployContext.BundlePushStatus.BundlePushStage {
+	// if the bundle push is already in progress, check the status
+	case enterpriseApi.BundlePushInProgress:
+		scopedLog.Info("checking the status of SHC Bundle Push")
+		// check if the bundle push is complete
+		ok, err := shcPlayBookContext.isBundlePushComplete()
+		if ok {
+			// set the bundle push status to complete
+			setBundlePushState(shcPlayBookContext.afwPipeline, enterpriseApi.BundlePushComplete)
+
+			// reset the retry count
+			shcPlayBookContext.afwPipeline.appDeployContext.BundlePushStatus.RetryCount = 0
+
+			// set the state to install complete for all the cluster scoped apps
+			setInstallStateForClusterScopedApps(appDeployContext, enterpriseApi.AppPkgInstallComplete)
+		} else if err != nil {
+			scopedLog.Error(err, "there was an error in SHC bundle push, will retry again.")
+			return err
```

Comment:
nit: `return` is redundant. Either if-else-if or explicit returns will avoid confusion.